### PR TITLE
Refactor github edit code

### PR DESCRIPTION
**Pull request by nick-mon1**
## Summary

Refactors `hugo` and `javascript` code sharing/coupling and removes `jquery` dependency to use native javascript features. This affects all templates that use 

### Preview

| Preview Link                                                                                                                                                                                                                                               | Functionality                          |
| ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------- |
| [Home](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-github-security/)                                                                                                              | Edit Items                    |
| [News](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-github-security/news/)                                                                                                         | Edit Items                    |
| [Events](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-github-security/events/)                                                                                                     | Edit Items                    |
| [Authors](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-github-security/authors/)                                                                                                   | Edit Items                    |
| [Event Page](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-github-security/2023/06/21/10-ways-to-recognize-employees-at-work/)                                                      | Edit Page (button in content footer) , Github Commit Info |
| [News Page](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-refactor-github-security/event/2023/05/24/strengthening-information-accessibility-for-consumers-with-limited-english-proficiency/) | Edit Page (button in content footer), Github Commit Info |

### Solution

1. **Template updates (layout component & partials).**
    1. remove hugo-javascript code sharing in `head.html` — 6d1d18e5ef4b00ac870fd5bff00d36c1e336b70d
    2. created `edit-this-button.html` partial — e725ff7c61f6fe65c998221330cb61d3ff6fb3b7
    3. renamed `page-data.html` to `gh-commit-info.html`
    4. removed unused code and renamed `core/page-data.html` to `core/gh-commit-info.html` — 70475395d802dce74e4e7b15f346e6ddeffb9450
    5. refactored class names to follow bem to be more consistent — 70475395d802dce74e4e7b15f346e6ddeffb9450
2. **Refactor JavaScript.**
    1. replaced `jquery` with native javascript [^1]
    2. renamed `edit-this.js` to `gh-edit-page.js` and `github.js` to `gh-edit-commit.js`
    3. added `jsdocs` to functions — 0c3242595d566aa7ebe1764ec3cc5be6a93580c3
    4. refactored `gh-edit-commit.js` by functionality — 6d267b2b42d29400b6e57dd012fa435c6edd5452



[^1]: **Note:** `jquery` node package dependency will be removed in [Refactoring images-stream to native javascript [PR#6421]](https://github.com/GSA/digitalgov.gov/pull/6421#top)


### How To Test

**Home/Landing Page**

1. Visit Home or any Landing Page
6. Click "Edit Tools" (floating blue pencil) button
7. Yellow highlight box is displayed and edit button links to github file

**Single Page**
1. Visit any news, events or single resource page
4. Click on "Edit Tools" button
6. Yellow highlight box is displayed and edit button links to github file
8. Bottom of page displays most recent commit date and links to github file

### Testing checklist

- [ ] Clicking edit button will show yellow edit sections.
- [ ] Edit links on highlighted sections link to correct file.
- [ ] Content footer on single pages shows recent commit date.
- [ ] Content footer on single pages correctly links to Github file.


### Screenshots

**Edit Button**

https://github.com/GSA/digitalgov.gov/assets/104778659/730c7910-9ec5-4147-88aa-637833055965

**Git Commit Information**

<img width="790" alt="Screen Shot 2023-08-04 at 3 11 56 PM" src="https://github.com/GSA/digitalgov.gov/assets/104778659/394cf870-d1e7-4a77-9a26-cdbe5d609e79">



---

### Dev Checklist

- [x] PR has correct labels
- [x] A11y testing (voice over testing, meets WCAG, run axe tools)
- [x] Code is formatted properly

### CODE REVIEW

File: themes/digital.gov/src/js/github.js [hunk 1]
```diff
@@ -1,89 +1,158 @@
...
-      $("#page-data .edit-file").html(edit).removeClass("hidden");
+/**
+ * Sets the most recent commit timestamp for the page
+ * This is displayed in #page-data section at the bottom of the page
+ */
+
+// eslint-disable-next-line func-names
+(function () {
+  const editFileButton = document.querySelector(".edit-file");
+  const editPageLink = document.querySelector("#page-data .edit-file");
```
Comment (mejiaj) [inline themes/digital.gov/src/js/github.js:9]: Can we use a consistent naming convention here? Something like `gh-edit-file` and `gh-edit-file-page-data` or `gh-edit-file-data`.

File: themes/digital.gov/src/js/github.js [hunk 2]
```diff
@@ -1,89 +1,158 @@
...
+  const gitRepo = {};
```
Comment (mejiaj) [inline themes/digital.gov/src/js/github.js:10]: We should export the shared settings to their own JS file and import that data in each script.

>**Warning**
> This functionality is added in #6419. So we can leave for now and refactor in the future.

File: themes/digital.gov/src/js/github.js [hunk 3]
```diff
@@ -1,89 +1,158 @@
...
+   * example string: news/2023/07/2023-07-19-gsa-shared-service-provider-program-guide.md
+   */
+  function getFilepath() {
+    if (editPageLink) {
+      gitRepo.filepath = document
+        .querySelector("div[data-edit-this]")
+        .getAttribute("data-edit-this");
+    } else {
+      gitRepo.filepath = "";
+    }
```
Comment (mejiaj) [inline themes/digital.gov/src/js/github.js:23]: You could just set `gitRepo.filepath` empty by default and override if it exists.

File: themes/digital.gov/src/js/github.js [hunk 4]
```diff
@@ -1,89 +1,158 @@
...
+   */
+  function getBranch() {
+    const host = window.location.hostname;
+    if (host.includes("sites.pages.cloud.gov")) {
+      // eslint-disable-next-line prefer-destructuring
+      gitRepo.branch = window.location.pathname.split("/")[4];
+    } else {
+      gitRepo.branch = "main";
+    }
+  }
```
Comment (mejiaj) [inline themes/digital.gov/src/js/github.js:39]: Similar to other script, maybe we can leverage the baseUrl.

File: themes/digital.gov/src/js/github.js [hunk 5]
```diff
@@ -1,89 +1,158 @@
...
+    return `${outputDate} at ${outputTime}`;
+  }
+
+  /**
+   * Add "Edit" link to the #page-data element before the "Last updated on" commit timestamp
+   * Checks if page has the #page-data element for single pages, not displayed on landing pages
+   */
+  function displayGithubEditLink() {
+    const gitEditFilePath = `https://github.com/GSA/digitalgov.gov/edit/${gitRepo.branch}/content/${gitRepo.filepath}`;
+    if (gitEditFilePath !== undefined) {
```
Comment (mejiaj) [inline themes/digital.gov/src/js/github.js:76]: If this is empty or `null` by default we can simplify this check.

```suggestion
    if (gitEditFilePath) {
```

**Review by mejiaj — CHANGES_REQUESTED** [review]
This is looking a lot better. Added suggestions that apply to both scripts.

---

**PS**

There's always some confusion in viewing `edit-this.js` and `github.js`, since they both involve GitHub.

We should consider renaming them and the class prefixes they depend on for clarity. For example using classes like `gh-inline-edit-this` or `gh-content-edit`.

File: themes/digital.gov/layouts/partials/core/head.html [hunk 6]
```diff
@@ -253,22 +253,6 @@
 
   <meta name="viewport" content="width=device-width,initial-scale=1" />
 
-  <!-- GitHub -->
-  <script type="text/javascript">
-    git_org          = {{ .Site.Params.git_org }};
-    git_repo         = {{ .Site.Params.git_repo }};
-    goUsaUsername    = {{ .Site.Params.goUsaUsername }};
-    goUsaApiKey      = {{ .Site.Params.goUsaApiKey }};
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/head.html:261]: They are not used anywhere else in the codebase, old code.

File: themes/digital.gov/layouts/partials/core/page-data.html [hunk 7]
```diff
@@ -1,17 +1,6 @@
...
-
-  - `.email` — for specifying a different primary email address to be used in feedback emails
-  - `.cc` — for specifying an additional email address to be added to the feedback emails
-  - `.page_type` — for specifying the page type (not being used at the moment)
-  - `.branch` — for specifying which branch the build is on. This helps the edit button to point to the right branch when viewing a preview page.
-*/}}
-
-{{- $email_copy := (print "We value your feedback and open questions. Thank you for helping to make our pages better.") -}}
-
-
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/page-data.html:13]: This is old code that is not used anymore for this template.

File: themes/digital.gov/src/js/edit-this.js [hunk 8]
```diff
@@ -1,52 +1,72 @@
...
-      const filepath = $(this).data("edit-this");
-      //
-      // Disabling no-undef & camelcase because these are defined in templates.
-      //
-      // eslint-disable-next-line no-undef, camelcase
-      const editLink = `<a class="edit-this-btn" href="https://github.com/${git_org}/${git_repo}/edit/${branch}/content/${filepath}" title="edit this" target="_blank"><span>edit</span></a>`;
+// Sets the github URL link and button styles for each resource when edit-mode is enabled
+// eslint-disable-next-line func-names
+(function () {
+  const githubEditLinks = document.querySelectorAll("*[data-edit-this]");
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/src/js/edit-this.js:4]: This is now removed, old selector from previous code.

File: themes/digital.gov/src/js/edit-this.js [hunk 9]
```diff
@@ -1,52 +1,72 @@
...
+  const editToolsButton = document.querySelector(".edit-tools");
+  const gitRepo = {};
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/src/js/edit-this.js:6]: Set `null` as the defaults in an object.

File: themes/digital.gov/src/js/edit-this.js [hunk 10]
```diff
@@ -1,52 +1,72 @@
...
+    gitRepo.branch = "main";
+  }
+
+  /**
+   * Adds github "button" link and highlight style to Github items
+   * Add link as last child in the parent element
+   */
+  function enableEditMode() {
+    githubEditLinks.forEach((link) => {
+      gitRepo.filepath = link.getAttribute("data-edit-this");
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/src/js/edit-this.js:27]: Agreed, updated with `dataset.editThis`.

File: themes/digital.gov/src/js/edit-this.js [hunk 11]
```diff
@@ -1,52 +1,72 @@
...
+      const editSpan = Object.assign(document.createElement("span"), {
+        innerHTML: "edit",
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/src/js/edit-this.js:29]: This is now updated to use an `a` tag.

File: themes/digital.gov/src/js/edit-this.js [hunk 12]
```diff
@@ -1,52 +1,72 @@
...
-
-    const $editIcon = $(this).find("i");
-    const editIconDefaultClass = "far fa-edit";
-    const editIconActiveClass = "fas fa-times";
+  /**
+   * Add event listener to the edit tools button in lower right corner
+   */
+  // eslint-disable-next-line func-names
+  editToolsButton.addEventListener("click", function (event) {
+    event.preventDefault();
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/src/js/edit-this.js:62]: Yes, I just tested this and removed it.

File: themes/digital.gov/layouts/partials/core/page-data.html [hunk 13]
```diff
@@ -1,17 +1,6 @@
...
-<div id="page-data" class="branch-{{- .branch -}}" name="page-data">
+<div id="page-data" class="branch" name="page-data">
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/page-data.html:3]: Good call, removed.

File: themes/digital.gov/layouts/partials/core/page-data.html [hunk 14]
```diff
@@ -1,17 +1,6 @@
...
   <!-- Edit file on GitHub -->
-  <div class="edit-file hidden"></div>
+  <div class="edit-file"></div>
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/page-data.html:5]: Good call on the `<noscript>`, I didn't know what this tag is used for until now. I've added a detailed comment.

File: themes/digital.gov/src/js/edit-this.js [hunk 15]
```diff
@@ -1,52 +1,72 @@
...
+   * If on cloud.pages then get the branch name from the URL
+   * Otherwise, use main for local host and production
+   */
+  const host = window.location.hostname;
+  if (host.includes("sites.pages.cloud.gov")) {
+    // eslint-disable-next-line prefer-destructuring
+    gitRepo.branch = window.location.pathname.split("/")[4];
+  } else {
+    gitRepo.branch = "main";
+  }
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/src/js/edit-this.js:19]: Good idea, I've made a note to discuss more on this.

File: themes/digital.gov/src/js/edit-this.js [hunk 16]
```diff
@@ -1,52 +1,72 @@
...
+   * Removes the last child link from the parent element
+   */
+  function disableEditMode() {
     // eslint-disable-next-line func-names
-    $("*[data-edit-this]").each(function () {
-      $(this).removeClass("edit-this");
-      $(".edit-this-btn").remove();
+    githubEditLinks.forEach((link) => {
+      link.classList.remove("edit-this");
+      link.lastChild.remove();
```
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/src/js/edit-this.js:53]: Hmmm, didn't think of that idea. We could put `visibility: hidden;` to hide them after the first time clicking the `edit-this-btn` or something like that. Created a trello card to look into this.

File: themes/digital.gov/src/js/github.js (hunk 1 above)
Comment (nick-mon1) [reply themes/digital.gov/src/js/github.js:9]: Yes, I've renamed the classes for all github editing related functions.

**Comment by mejiaj** [comment]
@nick-mon1 can you fix conflicts?

File: themes/digital.gov/layouts/partials/core/page-data.html [hunk 17]
```diff
@@ -1,17 +1,14 @@
...
+    @example Edit Last updated on Jun 13, 2023 at 5:45 p.m., ET
 */}}
-
-{{- $email_copy := (print "We value your feedback and open questions. Thank you for helping to make our pages better.") -}}
-
-
-<div id="page-data" class="branch-{{- .branch -}}" name="page-data">
-  <!-- Edit file on GitHub -->
-  <div class="edit-file hidden"></div>
+<!-- prettier-ignore-end -->
```
Comment (mejiaj) [inline themes/digital.gov/layouts/partials/core/page-data.html:9]: A few notes:
1. Indents look one level too far.
2. Updating to latest version of `prettier` & `prettier-plugin-go-template` has support for multiline comments, so you don't need to ignore. In general we should avoid ignoring as much as possible ― although I totally get why it's used here.

File: themes/digital.gov/src/js/gh-edit-commit.js [hunk 18]
```diff
@@ -0,0 +1,159 @@
+/**
+ * Sets the most recent commit timestamp for the page
+ * This is displayed in #page-data section at the bottom of the page
+ */
+
+// eslint-disable-next-line func-names
+(function () {
+  const editPageLink = document.querySelector("#page-data .gh-commit-date");
```
Comment (mejiaj) [inline themes/digital.gov/src/js/gh-edit-commit.js:8]: Curious about the specificity of this selector. Not against it, but couldn't we simply query `.gh-commit-date`? 

Are there multiple elements with this class?

File: themes/digital.gov/src/js/gh-edit-commit.js [hunk 19]
```diff
@@ -0,0 +1,159 @@
...
+  const gitRepo = {
+    filepath: "",
+    branch: null,
+  };
+
+  /**
+   * Get hugo file path from the data-gh-edit-page attribute to link to Github repo location
+   * example string: news/2023/07/2023-07-19-gsa-shared-service-provider-program-guide.md
```
Comment (mejiaj) [inline themes/digital.gov/src/js/gh-edit-commit.js:16]: You can use `@example` here.

```suggestion
   * @example
   * // returns string:
   * // news/2023/07/2023-07-19-gsa-shared-service-provider-program-guide.md
```

https://jsdoc.app/tags-example.html

File: themes/digital.gov/src/js/gh-edit-commit.js [hunk 20]
```diff
@@ -0,0 +1,159 @@
...
+   */
+  function getFilepath() {
+    if (editPageLink) {
+      gitRepo.filepath = document
+        .querySelector("div[data-gh-edit-page]")
```
Comment (mejiaj) [inline themes/digital.gov/src/js/gh-edit-commit.js:21]: Consider a less specific selector, if we wanted to move it to `article` or `main` this wouldn't work.

```suggestion
        .querySelector("[data-gh-edit-page]")
```

File: themes/digital.gov/src/js/gh-edit-commit.js [hunk 21]
```diff
@@ -0,0 +1,159 @@
...
+    if (host.includes("sites.pages.cloud.gov")) {
+      // eslint-disable-next-line prefer-destructuring
+      gitRepo.branch = window.location.pathname.split("/")[4];
+    } else {
+      gitRepo.branch = "main";
+    }
+  }
+
+  /**
+   * format github ISO date format to human friendly datetime string
```
Comment (mejiaj) [inline themes/digital.gov/src/js/gh-edit-commit.js:42]: Can we use full sentences for descriptions? Makes it easier to read.

File: themes/digital.gov/src/js/gh-edit-commit.js [hunk 22]
```diff
@@ -0,0 +1,159 @@
+/**
+ * Sets the most recent commit timestamp for the page
+ * This is displayed in #page-data section at the bottom of the page
+ */
+
+// eslint-disable-next-line func-names
+(function () {
```
Comment (mejiaj) [inline themes/digital.gov/src/js/gh-edit-commit.js:7]: A better way to avoid the eslint error is to use an arrow function.

```suggestion
(() => {
```

**Review by mejiaj — CHANGES_REQUESTED** [review]
This is coming along nicely. Added some comments to improve clarity & follow best practices.

File: themes/digital.gov/layouts/404.html
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/layouts/404.html:1]: Good catch

File: themes/digital.gov/layouts/communities/single.html
[the remark this answers is not in the document]
Comment (nick-mon1) [reply themes/digital.gov/layouts/communities/single.html:1]: Updated the template to have the header and content section with the edit fields. This is the convention that the other templates follow.

File: themes/digital.gov/src/js/gh-edit-commit.js (hunk 18 above)
Comment (nick-mon1) [reply themes/digital.gov/src/js/gh-edit-commit.js]: Shortened to just one selector, needed to be updated with the new class names.

File: themes/digital.gov/layouts/partials/core/page-data.html (hunk 17 above)
Comment (nick-mon1) [reply themes/digital.gov/layouts/partials/core/page-data.html:9]: Extra indents are fixed, good to know about the package updates, added to the to do list 👍🏻

File: themes/digital.gov/src/js/gh-edit-commit.js (hunk 19 above)
Comment (nick-mon1) [reply themes/digital.gov/src/js/gh-edit-commit.js]: Ahh, thanks for the proper usage format.

File: themes/digital.gov/src/js/gh-edit-commit.js (hunk 20 above)
Comment (nick-mon1) [reply themes/digital.gov/src/js/gh-edit-commit.js]: Great point, selector is updated!

File: themes/digital.gov/src/js/gh-edit-commit.js (hunk 21 above)
Comment (nick-mon1) [reply themes/digital.gov/src/js/gh-edit-commit.js:57]: Comment is updated to read better.

File: themes/digital.gov/src/scss/new/_gh-commit-info.scss [hunk 23]
```diff
@@ -0,0 +1,31 @@
...
+  &__edit-button {
+    @include u-font("sans", "xs");
+    background-color: white;
+    border: 2px color("blue-warm-50v") solid;
+    border-radius: 0.25rem;
+    color: color("blue-warm-50v");
+    font-weight: 600;
+    padding: units("2px") units(1);
+    text-decoration: none;
+  }
```
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/_gh-commit-info.scss:24]: We can get rid of these styles if we use the button outline variant.

The classes are: `.usa-button.usa-button--outline`.

File: themes/digital.gov/src/scss/new/_gh-edit-page.scss [hunk 24]
```diff
@@ -0,0 +1,145 @@
...
+    color: color("blue-50v");
+    padding: 12px;
+
+    &:hover {
+      background-color: color("gray-3");
+    }
+
+    &:focus {
+      background-color: color("gray-3");
+    }
```
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/_gh-edit-page.scss:122]: You could just combine these.

```suggestion
    &:hover,
    &:focus {
      background-color: color("gray-3");
    }
```

File: themes/digital.gov/src/scss/new/_gh-edit-page.scss [hunk 25]
```diff
@@ -0,0 +1,145 @@
...
+
+  .gh-edit-tools__issue-link {
+    @include u-font("sans", "lg");
+    background-color: color("red-40v");
+    border: units(1) color("red-50v") solid;
+    bottom: 110px;
+    color: white;
+    display: block;
+
+    padding: 12px;
```
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/_gh-edit-page.scss:133]: Remove extra line.

```suggestion
    display: block;
    padding: 12px;
```

File: themes/digital.gov/src/scss/new/_gh-edit-page.scss [hunk 26]
```diff
@@ -0,0 +1,145 @@
...
+
+    &:hover {
+      background-color: color("red-40v");
+      border: units(1) color("red-60v") solid;
+    }
+
+    &:focus {
+      background-color: color("red-40v");
+      border: units(1) color("red-60v") solid;
+    }
```
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/_gh-edit-page.scss:143]: Again, you can combine.

```suggestion
    &:hover,
    &:focus {
      background-color: color("red-40v");
      border: units(1) color("red-60v") solid;
    }
```

**Review by mejiaj — CHANGES_REQUESTED** [review]
Minor changes, otherwise LGTM.

File: themes/digital.gov/src/scss/new/_gh-commit-info.scss (hunk 23 above)
Comment (nick-mon1) [reply themes/digital.gov/src/scss/new/_gh-commit-info.scss:24]: Good call, removed styles and added `.usa-button.usa-button--outline`

**Pull request merged**